### ISSUE TITLE
Panic when summary divider is put on same line as a HTML element

### ISSUE THREAD

**Issue by Jos512** [spf13/hugo#2538]
Hello,

Using Hugo 0.16 (64bit) on Windows 7 SP1, I get the following when building the site:

```
PS I:\site\website> hugo server --buildDrafts
Started building site
1 of 1 draft rendered
0 future content
367 pages created
0 non-page files copied
0 paginator pages created
0 n1 created
0 n2 created
in 1095 ms
Watching for changes in I:\site\website\{data,content,layouts,static,themes}
Serving pages from memory
Web Server is available at http://localhost:1313/ (bind address 127.0.0.1)
Press Ctrl+C to stop
```

Now using Hugo 0.17 (64bit):

```
PS I:\site\website> hugo server --buildDrafts
Started building sites ...
panic: runtime error: slice bounds out of range

goroutine 777 [running]:
panic(0xa49d80, 0xc042004050)
        /usr/local/Cellar/go/1.7.1/libexec/src/runtime/panic.go:500 +0x1af
github.com/spf13/hugo/hugolib.splitUserDefinedSummaryAndContent(0xaebc08, 0x8, 0xc043886700, 0x1a4c, 0x1a4d, 0xc04327efc
0)
        /Users/spf13/gopath/src/github.com/spf13/hugo/hugolib/page.go:337 +0x8e1
github.com/spf13/hugo/hugolib.(*Page).setUserDefinedSummaryIfProvided(0xc043950a00, 0x0, 0x0, 0x0)
        /Users/spf13/gopath/src/github.com/spf13/hugo/hugolib/page.go:262 +0x6d
github.com/spf13/hugo/hugolib.(*Site).preparePagesForRender.func1(0xc043270101, 0x101, 0x0, 0xc04208eb00, 0xc04327efc0,
0xc043a85f50)
        /Users/spf13/gopath/src/github.com/spf13/hugo/hugolib/hugo_sites.go:482 +0x2e4
created by github.com/spf13/hugo/hugolib.(*Site).preparePagesForRender
        /Users/spf13/gopath/src/github.com/spf13/hugo/hugolib/hugo_sites.go:507 +0x10e
```

I don't know what causes the "slice bounds out of range" error -- I didn't change anything in my theme or configuration between building the site with Hugo 0.16 and Hugo 0.17. 

I also cannot find in the error message where I misused a function that might work correctly on 0.16 but not on 0.17.

Thanks,

*Edit*: This issue appears with another website than the issue I reported yesterday ([Hugo 0.17 requires taxonomy definition](https://github.com/spf13/hugo/issues/2534)). That means the "slice bounds out of range" error is not a global issue happening on each of my sites.

**Comment by bep** [comment]
You have added a summary divider () on a page in a way I didn't expect. As I didn't expect it it isn't possible for me to say what without seeing the full site source.

**Comment by Jos512** [comment]
> You have added a summary divider () on a page in a way I didn't expect. As I didn't expect it it isn't possible for me to say what without seeing the full site source.

Thanks Bep! I had four content files in which I used `<!---more-->`; after removing that statement, the issue went away and Hugo 0.17 build the site correctly.

I can replicate the issue by re-adding `<!--more-->` to my `.md` files that contain HTML code. For instance, this triggers the error:

```
<p class="lead"><strong>Situation</strong><br/>
This is my introduction.</p><!--more-->
```

If I place `<!--more-->` on a separate line following the HTML-formatted paragraph, there is **no** error:

```
<p class="lead"><strong>Situation</strong><br/>
This is my introduction.</p>

<!--more-->
```

Placing `<!--more-->` on the same line as a markdown-formatted paragraph like this **does not** trigger the error:

`*Situation*: This is my introduction.<!--more-->`

So to summarise, it seems Hugo doesn't expect `<!--more-->` to be placed on the same line of another HTML element.

(That probably makes sense. In my case, the error triggered on some legacy articles that I imported from WordPress and I didn't convert those to markdown.)

**Comment by bep** [comment]
Thanks for the detailed explanation -- we will fix this, sorry about the noise.

**Comment by Jos512** [comment]
No problem, I'm glad I can do my part to make Hugo a little bit better. Have a nice day!

**Comment by bep** [comment]
I'm having trouble reproducing this.

1) What is the output of `hugo version`? (is it the latest 0.17...?)
2) Could you provide the full content of a failing md-file?

**Comment by Jos512** [comment]
Hi Bep,

1) 0.17:

`Hugo Static Site Generator v0.17 BuildDate: 2016-10-07T14:45:26+02:00`

More specifically, I downloaded [this version](https://github.com/spf13/hugo/releases/download/v0.17/hugo_0.17_Windows-64bit.zip).

2) I've uploaded a failing md file to Dropbox [here](https://dl.dropboxusercontent.com/u/78986515/switch-fall-through.md).
